TF-3977 Add disable spam banner in preferences setting

dab246 opened the pull request:
## Issue

#3977 

## Resolved


https://github.com/user-attachments/assets/145142e0-5321-467a-a14f-c9f1271d6e79

dab246 committed:
TF-3977 Add disable spam banner in preferences setting

tddang-linagora reviewed (commented):
Same issue on Android

dab246 committed:
TF-3977 Setup disable thread as default

dab246 commented:
> * [x]  Thread displayed although thread setting is disabled
>

Fixed


https://github.com/user-attachments/assets/3d03b196-715e-41c0-94f1-1b220333da3c



> Same issue on Android

Fixed

[Screen_recording_20250903_131246.webm](https://github.com/user-attachments/assets/e74db920-cade-4a6f-a2d5-89420ec2dae2)

dab246 commented on lib/features/mailbox_dashboard/data/datasource_impl/local_spam_report_datasource_impl.dart:
```diff
@@ -5,35 +5,50 @@ import 'package:jmap_dart_client/jmap/core/user_name.dart';
 
 class LocalSpamReportDataSourceImpl extends SpamReportDataSource {
-  final LocalSpamReportManager _localSpamReportManager;
+  final PreferencesSettingManager _preferencesSettingManager;
```
Updated. 

<img width="802" height="55" alt="Screenshot 2025-09-09 at 11 02 07 AM" src="https://github.com/user-attachments/assets/14da67f7-9c75-4f66-be8a-256ecf1ff645" />

dab246 committed:
TF-3977 Separate settings keys for easier in conflict

hoangdat commented:
please rebase
